server/src/main/java/org/opensearch/ratelimitting/admissioncontrol/controllers/AdmissionController.java

```diff
@@ -54,17 +75,31 @@ public String getName() {
     }
 
     /**
-     * Adds the rejection count for the controller. Primarily used when copying controller states.
-     * @param count To add the value of the tracking resource object as the provided count
+     * Add rejection count to the rejection count metric tracked by the admission controller
      */
-    public void addRejectionCount(long count) {
-        this.rejectionCount.addAndGet(count);
+    public void addRejectionCount(String admissionControlActionType, long count) {
+        AtomicLong updatedCount = new AtomicLong(0);
+        if (this.rejectionCountMap.containsKey(admissionControlActionType)) {
+            updatedCount.addAndGet(this.rejectionCountMap.get(admissionControlActionType).get());
+        }
+        updatedCount.addAndGet(count);
+        this.rejectionCountMap.put(admissionControlActionType, updatedCount);
     }
 
     /**
      * @return current value of the rejection count metric tracked by the admission-controller.
      */
-    public long getRejectionCount() {
-        return this.rejectionCount.get();
+    public long getRejectionCount(String admissionControlActionType) {
+        AtomicLong rejectionCount = this.rejectionCountMap.getOrDefault(admissionControlActionType, new AtomicLong());
```

Comment:
Avoid `new` as much as possible :) It makes GC to do work and can affact tail latencies.
Let's return `0` in case no key is present in map.

Reply:
Agree will get this updated.